### PR TITLE
CIF-201 - Document CIF API Contract on public GH

### PR DESCRIPTION
**Pull request by dplaton**

### CODE REVIEW

File: documentation/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Adobe CIF cloud API
+
+## REST
+
+The Adobe CIF Cloud API follows the [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) principles which involve separating the APIs into logical resources that are manipulated using HTTP requests where the method (GET, POST, PUT, PATCH, DELETE) has specific meaning. As a consequence, the API can be consumed by all standard HTTP clients.
+
+## API versioning
```
[the remark this answers is not in the document]
Comment (herzog31) [reply documentation/README.md]: Maybe we can say that starting with the release `1.0` we use semantic versioning for actions. However, the API as exposed by the Runtime gateway is currently not versioned. As @cjelger said, the versioning is done as part of the deployment. Let's say we have shared actions available in version `1.0` and `1.1`, then the actual user of those actions can decide which ones to use and under which endpoint they should be exposed.
At some point we might want to define best practices though. That makes it easier for clients to consume the API.

File: documentation/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Content negociation
+
+Although the API will use only JSON for data serialization, it is good practice for clients to specify `Accept: application/json` in their request header.
+
+## Error codes
+
+The API uses standard HTTP response codes as error codes. Here's a brief rundown of the errors generated by the API:
+
+| Error code | Description |
```
[the remark this answers is not in the document]
Comment (herzog31) [reply documentation/README.md]: I agree, in general we should reference the API definition a bit more in this document.

File: documentation/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+The Adobe CIF Cloud API follows the [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) principles which involve separating the APIs into logical resources that are manipulated using HTTP requests where the method (GET, POST, PUT, PATCH, DELETE) has specific meaning. As a consequence, the API can be consumed by all standard HTTP clients.
+
+## Authorization
+
+TBD
+
+## JSON
+
+All response payloads are in JSON format.
```
Comment (masodu) [inline documentation/README.md]: Is this really correct? I think that this is more runtime/OW features so maybe we should just have link to that documentation.

File: documentation/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## Content negociation
+
+Although the API will use only JSON for data serialization, it is good practice for clients to specify `Accept: application/json` in their request header.
```
Comment (masodu) [inline documentation/README.md]: Same as for the above comment.

File: documentation/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+## Error codes
+
+The API uses standard HTTP response codes as error codes. For details about the error codes returned by specific endpoints you can check the [Swagger](http://opensource.adobe.com/commerce-cif-api/) documentation.
+
+## Paging and sorting
+
+The API methods that list entities accept parameters that control sorting and pagination. You can find more details in the [Swagger docs](http://opensource.adobe.com/commerce-cif-api/).
```
Comment (masodu) [inline documentation/README.md]: either update to Swagger docs the link at Error codes or remove docs from the above one.

File: documentation/README.md (hunk 3 above)
Comment (mhaack) [reply documentation/README.md]: I think this is correct for CIF at the moment we only support JSON format. OW supports more yes but this is CIF doc.

File: documentation/README.md (hunk 4 above)
Comment (mhaack) [reply documentation/README.md]: Same as above for CIF.

File: documentation/README.md (hunk 5 above)
Comment (dplaton) [reply documentation/README.md]: @masodu I'm not sure what you mean here. I couldn't find a way to link directly at the error codes for each function.

**Pull request merged**